services/api/src/resources/task/task_definition_resolvers.ts

```diff
@@ -309,6 +309,106 @@ export const addAdvancedTaskDefinition = async (
   );
 };
 
+export const updateAdvancedTaskDefinition = async (
+  {
+    input: {
+      id,
+      patch,
+      patch: {
+        name,
+        description,
+        image,
+        type,
+        service,
+        command,
+        project,
+        groupName,
+        environment,
+        permission,
+        created
+      }
+    }
+  },
+  { sqlClientPool, hasPermission, models, userActivityLogger }
+) => {
+  if (isPatchEmpty({ patch })) {
+    throw new Error('Input patch requires at least 1 attribute');
+  }
+
+  let projectObj = await getProjectByEnvironmentIdOrProjectId(
+    sqlClientPool,
+    environment,
+    project
+  );
+
+  const systemLevelTask =
```

Comment:
So I'm thinking about this part of the code. I think what we need to do is refactor this stuff and encapsulate a lot of this logic into either functions or classes.